checkers/always_returning_checker.py

```diff
@@ -0,0 +1,89 @@
+"""checker for always returning in a loop.
+"""
+
+import astroid
+from pylint.interfaces import IAstroidChecker
+from pylint.checkers import BaseChecker
+from pylint.checkers.utils import check_messages
+
+
+class AlwaysReturnChecker(BaseChecker):
+
+    __implements__ = IAstroidChecker
+
+    name = 'always_return'
+    msgs = {'E9996': ('Always returning an object or none in the loop on line %s',
+                      'always_returning_in_a_loop',
+                      'Used when you always return an object or none in a loop, '
+                      'this may cause the loop only runs once.'),
+           }
+
+    # this is important so that your checker is executed before others
+    priority = -1
+
+    @check_messages("always_returning_in_a_loop")
+    def helper_func(self, node, args):
+        # Check if the node is a return statement
+        if isinstance(node, astroid.Return):
+            self.add_message("always_returning_in_a_loop", node=node, args=args)
+        # check if the node a if statement
+        if isinstance(node, astroid.If):
+            # check the body of the if node
+            for child in node.body:
+                if isinstance(child, astroid.If):
+                    return self.helper_func(child, args)
+            # set the flag = True.
+            # Once there is not an explicit return, flag = false
+            flag = True
+            if not any([isinstance(child, astroid.Return) for child in (
+                    node.body + node.orelse)]):
+                flag = False
+            # set first_branch
+            if node.orelse and isinstance(node.orelse[0], astroid.If):
+                first_branch = node.orelse[0]
+            elif node.orelse:
+                # Return False if there is no return in else branch
+                if not any([isinstance(child, astroid.Return) for child in (
```

Comment:
@Jasminewwwu You should be making a recurisve call here, as well (and in the other places similar to this one).